db/upgrade.php

```diff
@@ -628,5 +628,17 @@ function xmldb_attendance_upgrade($oldversion=0) {
 
     }
 
+    if ($oldversion < 2019062300) {
+        // make entries in (calendar) event table compatible with timeline block
+        $lookbackdays = time() - 30 * (60 * 60 * 24); // no point changing all historical events on large sites
```

Comment:
This feels a little arbitrary to me.. although do we even need to deal with historical events? - the timeline block only shows "future" events right? - maybe we don't need to convert events older than time() at all?

Reply:
Agree the 30 days is arbitrary, perhaps a setting would be more appropriate? The timeline block can show "Overdue" events. I see a use case for teachers who have not marked the attendance for previous sessions.

Reply:
right - forgot about overdue events - using a setting to control upgrade scripts doesn't usually work too well, I'm ok with the arbitrary decision to only care about events older than 30 days for now - although it would be nice for you to also make sure the coding guideline warnings that will be showing up for those comments (and any other new changes) are also fixed and not just the errors... (check the travis log for all the new warnings that your code is creating) - this is particularly useful because the moodle.org plugins db reports on them too and I'm down to 6 warnings in the last version I pushed through.

one of the items that it will be picking up is that your comments should be starting with a capital letter and ending in a full stop... (annoying, but it does make the code clean when it follows guidelines!) :-)